File: mobile-app/services/notification.service.ts
// mobile-app/services/notification.service.ts
import apiClient from './api.client';

export type NotificationType = 'REGISTRATION_CONFIRMATION' | 'REMINDER' | 'CANCELLATION' | 'EMPLOYEE_REGISTERED_PARTICIPANT';

export type Notification = {
    id: number;
    eventId: number;
    eventTitle: string;
    notificationType: NotificationType;
    textContent: string;
    isRead: boolean;
    createdAt: string;
    readAt: string | null;
    eventStartDateTime: string | null;
    participantName: string | null;
};

export async function getNotifications(): Promise<Notification[]> {
    const response = await apiClient.get<Notification[]>('/notifications');
    return response.data;
}

export async function getUnreadCount(): Promise<number> {
    const response = await apiClient.get<{ count: number }>('/notifications/unread-count');
    return response.data.count;
}

export async function markAsRead(notificationId: number): Promise<Notification> {
    const response = await apiClient.put<Notification>(`/notifications/${notificationId}/read`);
    return response.data;
}

export async function markAllAsRead(): Promise<void> {
    await apiClient.put('/notifications/read-all');
}

export async function deleteNotification(notificationId: number): Promise<void> {
    await apiClient.delete(`/notifications/${notificationId}`);
}

File: openhand-backend/src/main/java/com/mana/openhand_backend/notifications/businesslayer/NotificationService.java
package com.mana.openhand_backend.notifications.businesslayer;

import com.mana.openhand_backend.notifications.dataaccesslayer.Notification;
import java.util.List;

public interface NotificationService {

    /**
     * Create a notification for a user
     * 
     * @param userId the user ID
     * @param eventId the event ID
     * @param notificationType the type of notification
     * @param language the user's preferred language
     * @return the created notification
     */
    Notification createNotification(Long userId, Long eventId, String notificationType, String language);

    /**
     * Create a notification for an employee/admin who registered someone else
     * 
     * @param userId the user ID (employee/admin)
     * @param eventId the event ID
     * @param notificationType the type of notification
     * @param language the user's preferred language
     * @param participantName the name of the person who was registered
     * @return the created notification
     */
    Notification createNotification(Long userId, Long eventId, String notificationType, String language, String participantName);

    /**
     * Get all notifications for a user, ordered by newest first
     */
    List<Notification> getUserNotifications(Long userId);
    /**
     * Create a donation notification for a user
     * 
     * @param userId the user ID
     * @param language the user's preferred language
     * @return the created donation notification
     */
    Notification createDonationNotification(Long userId, String language);

    /**
     * Get count of unread notifications for a user
     */
    long getUnreadCount(Long userId);

    /**
     * Mark a notification as read
     */
    Notification markAsRead(Long notificationId);

    /**
     * Mark all notifications as read for a user
     */
    void markAllAsRead(Long userId);

    /**
     * Delete a notification
     */
    void deleteNotification(Long notificationId);
}
